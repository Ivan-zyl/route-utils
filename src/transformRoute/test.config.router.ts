const list = [
  {
    path: '/home',
    component: './Home',
    menuKey: 'home',
    parentKey: '',
  },
  {
    path: '/manageSite',
    component: './platformConfig/ManageSite',
    permission: 'KM_PLAT_ADMIN',
    menuKey: 'manageSite',
    parentKey: 'platformConfig',
  },
  {
    path: '/manageSite/addSite',
    component: './platformConfig/AddSite',
    permission: 'KM_PLAT_ADMIN',
    menuKey: 'manageSite',
    parentKey: 'platformConfig',
  },
  {
    path: '/manageSite/updateSite/:id',
    component: './platformConfig/UpdateSite',
    permission: 'KM_PLAT_ADMIN',
    menuKey: 'manageSite',
    parentKey: 'platformConfig',
  },
  {
    path: '/manageSite/viewSite/:id',
    component: './platformConfig/UpdateSite',
    permission: 'KM_PLAT_ADMIN',
    menuKey: 'manageSite',
    parentKey: 'platformConfig',
  },
  /**
   * 金融云站点租户
   */
  {
    path: '/siteTenant',
    component: './platformConfig/SiteTenant',
    permission: 'KM_PLAT_ADMIN',
    menuKey: 'siteTenant',
    parentKey: 'platformConfig',
  },
  {
    path: '/siteTenant/addTenant',
    component: './platformConfig/AddTenant',
    permission: 'KM_PLAT_ADMIN',
    menuKey: 'siteTenant',
    parentKey: 'platformConfig',
  },
  {
    path: '/siteTenant/updateTenant/:id',
    component: './platformConfig/AddTenant',
    permission: 'KM_PLAT_ADMIN',
    menuKey: 'siteTenant',
    parentKey: 'platformConfig',
  },
  {
    path: '/siteTenant/viewTenant/:id',
    component: './platformConfig/AddTenant',
    permission: 'KM_PLAT_ADMIN',
    menuKey: 'siteTenant',
    parentKey: 'platformConfig',
  },

  /**
   * 安全工程师信息
   */
  {
    path: '/engineer',
    component: './platformConfig/EngineerInfo',
    permission: 'KM_PLAT_ADMIN',
    menuKey: 'engineer',
    parentKey: 'platformConfig',
  },
  /**
   * 密码技术团队人员信息
   */
  {
    path: '/team',
    component: './platformConfig/TeamMembers',
    permission: 'KM_PLAT_ADMIN',
    menuKey: 'team',
    parentKey: 'platformConfig',
  },
  /**
   * 业务安全目标
   */

  {
    path: '/securityGoals',
    component: './platformConfig/SecurityGoals',
    permission: 'KM_PLAT_ADMIN',
    menuKey: 'securityGoals',
    parentKey: 'platformConfig',
  },
  {
    path: '/securityGoals/add',
    component: './platformConfig/AddSecurityGoals',
    permission: 'KM_PLAT_ADMIN',
    menuKey: 'securityGoals',
    parentKey: 'platformConfig',
  },
  {
    path: '/securityGoals/update/:id',
    component: './platformConfig/AddSecurityGoals',
    permission: 'KM_PLAT_ADMIN',
    menuKey: 'securityGoals',
    parentKey: 'platformConfig',
  },

  /**
   * 密钥算法
   */

  {
    path: '/algorithm',
    component: './platformConfig/Algorithm',
    permission: 'KM_PLAT_ADMIN',
    menuKey: 'algorithm',
    parentKey: 'platformConfig',
  },
  {
    path: '/algorithm/add',
    component: './platformConfig/AddAlgorithm',
    permission: 'KM_PLAT_ADMIN',
    menuKey: 'algorithm',
    parentKey: 'platformConfig',
  },
  {
    path: '/algorithm/update/:id',
    component: './platformConfig/AddAlgorithm',
    permission: 'KM_PLAT_ADMIN',
    menuKey: 'algorithm',
    parentKey: 'platformConfig',
  },
  /**
   * 平台配置
   */
  {
    path: '/createKey',
    component: './keyMangement/CreateKey',
    permission: 'KM_WORKFLOW',
    menuKey: 'createKey',
    parentKey: 'keyMangement',
  },
  {
    path: '/useTheExistingKey',
    component: './keyMangement/CreateKey',
    permission: 'KM_WORKFLOW',
    menuKey: 'useTheExistingKey',
    parentKey: 'keyMangement',
  },
  {
    path: '/queryPublicKeyContent',
    component: './keyMangement/QueryKeyContent',
    permission: 'KM_WORKFLOW',
    menuKey: 'queryPublicKeyContent',
    parentKey: 'keyMangement',
  },
  {
    path: '/queryKeyContent',
    component: './keyMangement/QueryKeyContent',
    permission: 'KM_WORKFLOW',
    menuKey: 'queryKeyContent',
    parentKey: 'keyMangement',
  },
  {
    path: '/queryKeyAttr',
    component: './keyMangement/QueryKeyAttr',
    permission: 'KM_WORKFLOW',
    menuKey: 'queryKeyAttr',
    parentKey: 'keyMangement',
  },
  {
    path: '/updateKeyAttr',
    component: './keyMangement/QueryKeyAttr',
    permission: 'KM_WORKFLOW',
    menuKey: 'updateKeyAttr',
    parentKey: 'keyMangement',
  },
  /**
   * 密码服务场景接入
   */
  {
    path: '/sceneAccess',
    component: './scene/SceneAccess',
    permission: 'KM_WORKFLOW',
    menuKey: 'sceneAccess',
    parentKey: 'scene',
  },

  /**
   * 小工具
   */
  {
    path: '/kmiTool',
    component: './keyMangement/KmiTool',
    permission: 'KM_WORKFLOW',
    menuKey: 'kmiTool',
    parentKey: 'tool',
  },
  {
    path: '/smallTool',
    component: './keyMangement/SmallTool',
    permission: 'KM_WORKFLOW',
    menuKey: 'smallTool',
    parentKey: 'tool',
  },
  /**
   * 我的需求单
   */
  {
    path: '/myDemandList',
    component: './demand/MyDemandList',
    permission: 'KM_WORKFLOW',
    menuKey: 'myDemandList',
    parentKey: 'demand',
  },
  {
    path: '/myDemandList/demandDetail/:id',
    component: './demand/DemandDetail',
    permission: 'KM_WORKFLOW',
    menuKey: 'myDemandList',
    parentKey: 'demand',
  },
  {
    path: '/temporaryDemand',
    component: './demand/TemporaryDemand',
    permission: 'KM_WORKFLOW',
    menuKey: 'temporaryDemand',
    parentKey: 'demand',
  },
  {
    path: '/temporaryDemand/detail/:id',
    component: './demand/TemporaryDemandDetail',
    permission: 'KM_WORKFLOW',
    menuKey: 'temporaryDemand',
    parentKey: 'demand',
  },
  /**
   * 我的审批
   */
  {
    path: '/myApproval',
    component: './approval/MyApproval',
    permission: 'KM_WORKFLOW',
    menuKey: 'myApproval',
    parentKey: 'approval',
  },
  {
    path: '/myApproval/approvalDetail/:id',
    component: './approval/ApprovalDetail',
    permission: 'KM_WORKFLOW',
    menuKey: 'myApproval',
    parentKey: 'approval',
  },
  {
    path: '/haveBeenApproval',
    component: './approval/MyApproval',
    permission: 'KM_WORKFLOW',
    menuKey: 'haveBeenApproval',
    parentKey: 'approval',
  },
  {
    path: '/haveBeenApproval/detail/:id',
    component: './approval/ApprovalDetail',
    permission: 'KM_WORKFLOW',
    menuKey: 'haveBeenApproval',
    parentKey: 'approval',
  },
  {
    path: '/claimed',
    component: './approval/DemandClaimed',
    permission: 'KM_WORKFLOW',
    menuKey: 'claimed',
    parentKey: 'approval',
  },
  {
    path: '/claimed/detail/:id',
    component: './approval/ApprovalDetail',
    permission: 'KM_WORKFLOW',
    menuKey: 'claimed',
    parentKey: 'approval',
  },
  {
    path: '/keyManageOfManager',
    component: './keyManageOfManager/KeyManageOfManager',
    permission: 'KM_ADMIN',
    menuKey: 'keyManageOfManager',
  },
  {
    path: '/keyManageOfManager/detail/:id',
    component: './demand/DemandDetail',
    menuKey: 'keyManageOfManager',
    parentKey: '',
  },
  /**
   * secretmng
   */
  {
    path: '/secretmng/commission/secretList',
    component: './secretmng/secretList',
    permission: 'KM_SECRET_ADMIN',
    menuKey: 'secretList',
    parentKey: 'subMenu_manage',
  },
  {
    path: '/secretmng/commission/manualAddSecret',
    component: './secretmng/manualAddSecret',
    permission: 'KM_SECRET_ADMIN',
    menuKey: 'manualAddSecret',
    parentKey: 'subMenu_manage',
  },
  {
    path: '/secretmng/commission/createTask',
    component: './secretmng/createTask',
    permission: 'KM_SECRET_ADMIN',
    menuKey: 'createTask',
    parentKey: 'subMenu_manage',
  },
  {
    path: '/secretmng/commission/componentBls',
    component: './secretmng/componentBls',
    permission: 'KM_SECRET_ADMIN',
    menuKey: 'componentBls',
    parentKey: 'subMenu_manage',
  },
  {
    path: '/secretmng/rotation/taskList',
    component: './secretmng/taskList',
    permission: 'KM_SECRET_ADMIN',
    menuKey: 'taskList',
    parentKey: 'subMenu_rotate',
  },
  {
    path: '/secretmng/rotation/alterPloyManagement',
    component: './secretmng/alterPloyManagement',
    permission: 'KM_SECRET_ADMIN',
    menuKey: 'alterPloyManagement',
    parentKey: 'subMenu_rotate',
  },
  {
    path: '/secretmng/rotation/alterPloyManagement/createAlterPloy',
    component: './secretmng/alterPloyManagement/components/createAlterPloy',
    permission: 'KM_SECRET_ADMIN',
    menuKey: 'createAlterPloy',
    parentKey: 'subMenu_rotate',
  },
  {
    path: '/secretmng/rotation/alterPloyManagement/editAlterPloy',
    component: './secretmng/alterPloyManagement/components/createAlterPloy',
    permission: 'KM_SECRET_ADMIN',
    menuKey: 'editAlterPloy',
    parentKey: 'subMenu_rotate',
  },
  {
    path: '/secretmng/rotation/mailPloyManagement',
    component: './secretmng/mailPloyManagement',
    permission: 'KM_SECRET_ADMIN',
    menuKey: 'mailPloyManagement',
    parentKey: 'subMenu_rotate',
  },
  {
    path: '/secretmng/rotation/mailPloyManagement/createMailPloy',
    component: './secretmng/mailPloyManagement/components/createMailPloy',
    permission: 'KM_SECRET_ADMIN',
    menuKey: 'createMailPloy',
    parentKey: 'subMenu_rotate',
  },
  {
    path: '/secretmng/rotation/mailPloyManagement/editMailPloy',
    component: './secretmng/mailPloyManagement/components/createMailPloy',
    permission: 'KM_SECRET_ADMIN',
    menuKey: 'editMailPloy',
    parentKey: 'subMenu_rotate',
  },
  {
    path: '/secretmng/rotation/blackList',
    component: './secretmng/blackList',
    permission: 'KM_SECRET_ADMIN',
    menuKey: 'blackList',
    parentKey: 'subMenu_rotate',
  },
];

export const routeList = list;
export default {
  path: '/',
  component: '../layout',
  indexRoute: { component: './Home' },
  childRoutes: list,
};
