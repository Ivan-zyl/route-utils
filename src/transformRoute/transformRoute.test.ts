import transformRoute from './transformRoute';

const routes = [
  {
    path: '/user',
    layout: false,
    routes: [
      {
        name: 'login',
        path: '/user/login',
        layout: false,
        component: './user/login',
      },
    ],
  },
  {
    path: '/welcome',
    name: 'welcome',
    icon: 'smile',
    component: './Welcome',
  },
  {
    path: '/admin',
    name: 'admin',
    icon: 'crown',
    access: 'canAdmin',
    component: './Admin',
    menu: { hideChildren: true },
    routes: [
      {
        path: '/admin/sub-page',
        name: 'sub-page',
        icon: 'smile',
        component: './Welcome',
      },
    ],
  },
  {
    path: '/admin2',
    name: 'admin',
    icon: 'crown',
    access: 'canAdmin',
    component: './Admin',
    hideInMenu: true,
    routes: [
      {
        path: '/admin/sub-page2',
        name: 'sub-page',
        icon: 'smile',
        component: './Welcome',
      },
    ],
  },
  {
    name: 'list.table-list',
    icon: 'table',
    path: '/list',
    component: './ListTableList',
  },
  {
    path: '/',
    name: 'home',
    component: './ListTableList',
  },
  {
    component: './404',
  },
];

test('normal', () => {
  const { menuData, breadcrumb } = transformRoute(routes, true, ({ id }) => {
    if (id === 'menu.list.table-list') return '查询表格';
    if (id === 'menu.admin') return '管理页';
    if (id === 'menu.admin.sub-page') return '二级管理页';
    if (id === 'menu.welcome') return '欢迎';
    return id;
  });
  expect(menuData).toMatchSnapshot();
  expect(breadcrumb).toMatchSnapshot();
});

test('normal ignoreFilter', () => {
  const { menuData, breadcrumb } = transformRoute(
    routes,
    true,
    ({ id }) => {
      if (id === 'menu.list.table-list') return '查询表格';
      if (id === 'menu.admin') return '管理页';
      if (id === 'menu.admin.sub-page') return '二级管理页';
      if (id === 'menu.welcome') return '欢迎';
      return id;
    },
    true,
  );
  expect(menuData).toMatchSnapshot();
  expect(breadcrumb).toMatchSnapshot();
});

test('disable locale', () => {
  const { menuData, breadcrumb } = transformRoute(
    routes,
    false,
    ({ id }) => {
      if (id === 'menu.list.table-list') return '查询表格';
      if (id === 'menu.admin') return '管理页';
      if (id === 'menu.admin.sub-page') return '二级管理页';
      if (id === 'menu.welcome') return '欢迎';
      return id;
    },
    true,
  );
  expect(menuData).toMatchSnapshot();
  expect(breadcrumb).toMatchSnapshot();
});

test('do not start with "/"', () => {
  const { menuData: userMenuData, breadcrumb } = transformRoute(
    [
      {
        name: '人员组织管理',
        path: 'admin/userMng/',
        children: [
          { name: '人员管理', path: '/admin/userMng/users' },
          { name: '单位管理', path: '/admin/userMng/companies' },
        ],
      },
    ],
    false,
  );
  expect(userMenuData).toMatchSnapshot();
  expect(breadcrumb).toMatchSnapshot();
});
