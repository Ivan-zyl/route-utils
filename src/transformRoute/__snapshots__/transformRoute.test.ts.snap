// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`disable locale 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "component": "./user/login",
        "key": "/user/login",
        "layout": false,
        "locale": "menu.login",
        "name": "login",
        "path": "/user/login",
        "pro_layout_parentKeys": Array [
          "/user",
        ],
        "routes": null,
      },
    ],
    "key": "/user",
    "layout": false,
    "locale": false,
    "path": "/user",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
  Object {
    "component": "./Welcome",
    "icon": "smile",
    "key": "/welcome",
    "locale": "menu.welcome",
    "name": "welcome",
    "path": "/welcome",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
  Object {
    "access": "canAdmin",
    "component": "./Admin",
    "icon": "crown",
    "key": "/admin",
    "locale": "menu.admin",
    "menu": Object {
      "hideChildren": true,
    },
    "name": "admin",
    "path": "/admin",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
  Object {
    "access": "canAdmin",
    "children": Array [
      Object {
        "access": "canAdmin",
        "component": "./Welcome",
        "hideInMenu": true,
        "icon": "smile",
        "key": "/admin/sub-page2",
        "locale": "menu.admin.sub-page",
        "name": "sub-page",
        "path": "/admin/sub-page2",
        "pro_layout_parentKeys": Array [
          "/admin2",
        ],
        "routes": null,
      },
    ],
    "component": "./Admin",
    "hideInMenu": true,
    "icon": "crown",
    "key": "/admin2",
    "locale": "menu.admin",
    "name": "admin",
    "path": "/admin2",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
  Object {
    "component": "./ListTableList",
    "icon": "table",
    "key": "/list",
    "locale": "menu.list.table-list",
    "name": "list.table-list",
    "path": "/list",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
  Object {
    "component": "./ListTableList",
    "key": "home",
    "locale": "menu.home",
    "name": "home",
    "path": "/",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
]
`;

exports[`disable locale 2`] = `
Map {
  "/user/login" => Object {
    "component": "./user/login",
    "key": "/user/login",
    "layout": false,
    "locale": "menu.login",
    "name": "login",
    "path": "/user/login",
    "pro_layout_parentKeys": Array [
      "/user",
    ],
    "routes": null,
  },
  "/user" => Object {
    "children": Array [
      Object {
        "component": "./user/login",
        "key": "/user/login",
        "layout": false,
        "locale": "menu.login",
        "name": "login",
        "path": "/user/login",
        "pro_layout_parentKeys": Array [
          "/user",
        ],
        "routes": null,
      },
    ],
    "key": "/user",
    "layout": false,
    "locale": false,
    "path": "/user",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
  "/welcome" => Object {
    "component": "./Welcome",
    "icon": "smile",
    "key": "/welcome",
    "locale": "menu.welcome",
    "name": "welcome",
    "path": "/welcome",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
  "/admin" => Object {
    "access": "canAdmin",
    "component": "./Admin",
    "icon": "crown",
    "key": "/admin",
    "locale": "menu.admin",
    "menu": Object {
      "hideChildren": true,
    },
    "name": "admin",
    "path": "/admin",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
  "/admin/sub-page2" => Object {
    "access": "canAdmin",
    "component": "./Welcome",
    "hideInMenu": true,
    "icon": "smile",
    "key": "/admin/sub-page2",
    "locale": "menu.admin.sub-page",
    "name": "sub-page",
    "path": "/admin/sub-page2",
    "pro_layout_parentKeys": Array [
      "/admin2",
    ],
    "routes": null,
  },
  "/admin2" => Object {
    "access": "canAdmin",
    "children": Array [
      Object {
        "access": "canAdmin",
        "component": "./Welcome",
        "hideInMenu": true,
        "icon": "smile",
        "key": "/admin/sub-page2",
        "locale": "menu.admin.sub-page",
        "name": "sub-page",
        "path": "/admin/sub-page2",
        "pro_layout_parentKeys": Array [
          "/admin2",
        ],
        "routes": null,
      },
    ],
    "component": "./Admin",
    "hideInMenu": true,
    "icon": "crown",
    "key": "/admin2",
    "locale": "menu.admin",
    "name": "admin",
    "path": "/admin2",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
  "/list" => Object {
    "component": "./ListTableList",
    "icon": "table",
    "key": "/list",
    "locale": "menu.list.table-list",
    "name": "list.table-list",
    "path": "/list",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
  "/" => Object {
    "component": "./ListTableList",
    "key": "home",
    "locale": "menu.home",
    "name": "home",
    "path": "/",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
}
`;

exports[`do not start with "/" 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": undefined,
        "key": "/admin/userMng/users",
        "locale": "menu.人员组织管理.人员管理",
        "name": "人员管理",
        "path": "/admin/userMng/users",
        "pro_layout_parentKeys": Array [
          "/admin/userMng/",
        ],
        "routes": null,
      },
      Object {
        "children": undefined,
        "key": "/admin/userMng/companies",
        "locale": "menu.人员组织管理.单位管理",
        "name": "单位管理",
        "path": "/admin/userMng/companies",
        "pro_layout_parentKeys": Array [
          "/admin/userMng/",
        ],
        "routes": null,
      },
    ],
    "key": "admin/userMng/",
    "locale": "menu.人员组织管理",
    "name": "人员组织管理",
    "path": "/admin/userMng/",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
]
`;

exports[`do not start with "/" 2`] = `
Map {
  "/admin/userMng/users" => Object {
    "key": "/admin/userMng/users",
    "locale": "menu.人员组织管理.人员管理",
    "name": "人员管理",
    "path": "/admin/userMng/users",
    "pro_layout_parentKeys": Array [
      "/admin/userMng/",
    ],
    "routes": null,
  },
  "/admin/userMng/companies" => Object {
    "key": "/admin/userMng/companies",
    "locale": "menu.人员组织管理.单位管理",
    "name": "单位管理",
    "path": "/admin/userMng/companies",
    "pro_layout_parentKeys": Array [
      "/admin/userMng/",
    ],
    "routes": null,
  },
  "/admin/userMng/" => Object {
    "children": Array [
      Object {
        "key": "/admin/userMng/users",
        "locale": "menu.人员组织管理.人员管理",
        "name": "人员管理",
        "path": "/admin/userMng/users",
        "pro_layout_parentKeys": Array [
          "/admin/userMng/",
        ],
        "routes": null,
      },
      Object {
        "key": "/admin/userMng/companies",
        "locale": "menu.人员组织管理.单位管理",
        "name": "单位管理",
        "path": "/admin/userMng/companies",
        "pro_layout_parentKeys": Array [
          "/admin/userMng/",
        ],
        "routes": null,
      },
    ],
    "key": "admin/userMng/",
    "locale": "menu.人员组织管理",
    "name": "人员组织管理",
    "path": "/admin/userMng/",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
}
`;

exports[`normal 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": undefined,
        "component": "./user/login",
        "key": "/user/login",
        "layout": false,
        "locale": "menu.login",
        "name": "menu.login",
        "path": "/user/login",
        "pro_layout_parentKeys": Array [
          "/user",
        ],
        "routes": null,
      },
    ],
    "key": "/user",
    "layout": false,
    "locale": false,
    "path": "/user",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
  Object {
    "children": undefined,
    "component": "./Welcome",
    "icon": "smile",
    "key": "/welcome",
    "locale": "menu.welcome",
    "name": "欢迎",
    "path": "/welcome",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
  Object {
    "access": "canAdmin",
    "children": undefined,
    "component": "./Admin",
    "icon": "crown",
    "key": "/admin",
    "locale": "menu.admin",
    "menu": Object {
      "hideChildren": true,
    },
    "name": "管理页",
    "path": "/admin",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
  Object {
    "children": undefined,
    "component": "./ListTableList",
    "icon": "table",
    "key": "/list",
    "locale": "menu.list.table-list",
    "name": "查询表格",
    "path": "/list",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
  Object {
    "children": undefined,
    "component": "./ListTableList",
    "key": "home",
    "locale": "menu.home",
    "name": "menu.home",
    "path": "/",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
]
`;

exports[`normal 2`] = `
Map {
  "/user/login" => Object {
    "component": "./user/login",
    "key": "/user/login",
    "layout": false,
    "locale": "menu.login",
    "name": "menu.login",
    "path": "/user/login",
    "pro_layout_parentKeys": Array [
      "/user",
    ],
    "routes": null,
  },
  "/user" => Object {
    "children": Array [
      Object {
        "component": "./user/login",
        "key": "/user/login",
        "layout": false,
        "locale": "menu.login",
        "name": "menu.login",
        "path": "/user/login",
        "pro_layout_parentKeys": Array [
          "/user",
        ],
        "routes": null,
      },
    ],
    "key": "/user",
    "layout": false,
    "locale": false,
    "path": "/user",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
  "/welcome" => Object {
    "component": "./Welcome",
    "icon": "smile",
    "key": "/welcome",
    "locale": "menu.welcome",
    "name": "欢迎",
    "path": "/welcome",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
  "/admin" => Object {
    "access": "canAdmin",
    "component": "./Admin",
    "icon": "crown",
    "key": "/admin",
    "locale": "menu.admin",
    "menu": Object {
      "hideChildren": true,
    },
    "name": "管理页",
    "path": "/admin",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
  "/admin/sub-page2" => Object {
    "access": "canAdmin",
    "component": "./Welcome",
    "hideInMenu": true,
    "icon": "smile",
    "key": "/admin/sub-page2",
    "locale": "menu.admin.sub-page",
    "name": "二级管理页",
    "path": "/admin/sub-page2",
    "pro_layout_parentKeys": Array [
      "/admin2",
    ],
    "routes": null,
  },
  "/admin2" => Object {
    "access": "canAdmin",
    "children": Array [
      Object {
        "access": "canAdmin",
        "component": "./Welcome",
        "hideInMenu": true,
        "icon": "smile",
        "key": "/admin/sub-page2",
        "locale": "menu.admin.sub-page",
        "name": "二级管理页",
        "path": "/admin/sub-page2",
        "pro_layout_parentKeys": Array [
          "/admin2",
        ],
        "routes": null,
      },
    ],
    "component": "./Admin",
    "hideInMenu": true,
    "icon": "crown",
    "key": "/admin2",
    "locale": "menu.admin",
    "name": "管理页",
    "path": "/admin2",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
  "/list" => Object {
    "component": "./ListTableList",
    "icon": "table",
    "key": "/list",
    "locale": "menu.list.table-list",
    "name": "查询表格",
    "path": "/list",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
  "/" => Object {
    "component": "./ListTableList",
    "key": "home",
    "locale": "menu.home",
    "name": "menu.home",
    "path": "/",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
}
`;

exports[`normal ignoreFilter 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "component": "./user/login",
        "key": "/user/login",
        "layout": false,
        "locale": "menu.login",
        "name": "menu.login",
        "path": "/user/login",
        "pro_layout_parentKeys": Array [
          "/user",
        ],
        "routes": null,
      },
    ],
    "key": "/user",
    "layout": false,
    "locale": false,
    "path": "/user",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
  Object {
    "component": "./Welcome",
    "icon": "smile",
    "key": "/welcome",
    "locale": "menu.welcome",
    "name": "欢迎",
    "path": "/welcome",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
  Object {
    "access": "canAdmin",
    "component": "./Admin",
    "icon": "crown",
    "key": "/admin",
    "locale": "menu.admin",
    "menu": Object {
      "hideChildren": true,
    },
    "name": "管理页",
    "path": "/admin",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
  Object {
    "access": "canAdmin",
    "children": Array [
      Object {
        "access": "canAdmin",
        "component": "./Welcome",
        "hideInMenu": true,
        "icon": "smile",
        "key": "/admin/sub-page2",
        "locale": "menu.admin.sub-page",
        "name": "二级管理页",
        "path": "/admin/sub-page2",
        "pro_layout_parentKeys": Array [
          "/admin2",
        ],
        "routes": null,
      },
    ],
    "component": "./Admin",
    "hideInMenu": true,
    "icon": "crown",
    "key": "/admin2",
    "locale": "menu.admin",
    "name": "管理页",
    "path": "/admin2",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
  Object {
    "component": "./ListTableList",
    "icon": "table",
    "key": "/list",
    "locale": "menu.list.table-list",
    "name": "查询表格",
    "path": "/list",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
  Object {
    "component": "./ListTableList",
    "key": "home",
    "locale": "menu.home",
    "name": "menu.home",
    "path": "/",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
]
`;

exports[`normal ignoreFilter 2`] = `
Map {
  "/user/login" => Object {
    "component": "./user/login",
    "key": "/user/login",
    "layout": false,
    "locale": "menu.login",
    "name": "menu.login",
    "path": "/user/login",
    "pro_layout_parentKeys": Array [
      "/user",
    ],
    "routes": null,
  },
  "/user" => Object {
    "children": Array [
      Object {
        "component": "./user/login",
        "key": "/user/login",
        "layout": false,
        "locale": "menu.login",
        "name": "menu.login",
        "path": "/user/login",
        "pro_layout_parentKeys": Array [
          "/user",
        ],
        "routes": null,
      },
    ],
    "key": "/user",
    "layout": false,
    "locale": false,
    "path": "/user",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
  "/welcome" => Object {
    "component": "./Welcome",
    "icon": "smile",
    "key": "/welcome",
    "locale": "menu.welcome",
    "name": "欢迎",
    "path": "/welcome",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
  "/admin" => Object {
    "access": "canAdmin",
    "component": "./Admin",
    "icon": "crown",
    "key": "/admin",
    "locale": "menu.admin",
    "menu": Object {
      "hideChildren": true,
    },
    "name": "管理页",
    "path": "/admin",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
  "/admin/sub-page2" => Object {
    "access": "canAdmin",
    "component": "./Welcome",
    "hideInMenu": true,
    "icon": "smile",
    "key": "/admin/sub-page2",
    "locale": "menu.admin.sub-page",
    "name": "二级管理页",
    "path": "/admin/sub-page2",
    "pro_layout_parentKeys": Array [
      "/admin2",
    ],
    "routes": null,
  },
  "/admin2" => Object {
    "access": "canAdmin",
    "children": Array [
      Object {
        "access": "canAdmin",
        "component": "./Welcome",
        "hideInMenu": true,
        "icon": "smile",
        "key": "/admin/sub-page2",
        "locale": "menu.admin.sub-page",
        "name": "二级管理页",
        "path": "/admin/sub-page2",
        "pro_layout_parentKeys": Array [
          "/admin2",
        ],
        "routes": null,
      },
    ],
    "component": "./Admin",
    "hideInMenu": true,
    "icon": "crown",
    "key": "/admin2",
    "locale": "menu.admin",
    "name": "管理页",
    "path": "/admin2",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
  "/list" => Object {
    "component": "./ListTableList",
    "icon": "table",
    "key": "/list",
    "locale": "menu.list.table-list",
    "name": "查询表格",
    "path": "/list",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
  "/" => Object {
    "component": "./ListTableList",
    "key": "home",
    "locale": "menu.home",
    "name": "menu.home",
    "path": "/",
    "pro_layout_parentKeys": Array [],
    "routes": null,
  },
}
`;
